pyproject.toml

```diff
@@ -12,6 +12,7 @@ dependencies = [
     "torch>=2.2.0",
     "lightning==2.3.0.dev20240328",
     "jsonargparse[signatures]>=4.27.6",
+    "litserve==0.0.0.dev2",   # imported by litgpt.deploy
```

Comment:
When I remember correctly, I had problems with tugging the import so that it doesn't always gets called. I think that was due to how we use inheritance in 

https://github.com/Lightning-AI/litgpt/blob/57f0a6105049cee68443ac2f763cad4979807c78/litgpt/deploy/serve.py#L18

But I suppose one can define the class in a function to avoid the import at the top. But that would be ugly. What do you think?

Reply:
You can follow this pattern: https://github.com/Lightning-AI/litgpt/blob/43c44323b9b7fb6e3b567c800e6668c3a9b1ecee/litgpt/data/prepare_starcoder.py#L8-L20

And then inside the `__init__()`, before calling `super().__init__()` check that `_LITSERVE_AVAILABLE` is True (this part is missing from the code I linked)